src/biz/storage.go

```diff
@@ -0,0 +1,64 @@
+package biz
+
+import (
+	"fmt"
+
+	"github.com/docker/docker/api/types"
+	humanize "github.com/dustin/go-humanize"
+)
+
+// DeviceMapperStats contains the stats of devicemapper storages driver type
+type DeviceMapperStats struct {
+	DataAvailable        uint64
+	DataUsed             uint64
+	DataTotal            uint64
+	DataUsagePercent     float64
+	MetadataAvailable    uint64
+	MetadataUsed         uint64
+	MetadataTotal        uint64
+	MetadataUsagePercent float64
+}
+
+// ParseDeviceMapperStats parses the output from the info DriverStatus info Array. This information is not consider stable by
+// Docker and could change anytime according to API the docs.
+func ParseDeviceMapperStats(info types.Info) (*DeviceMapperStats, error) {
+
+	if info.Driver != "devicemapper" {
+		return nil, fmt.Errorf("only devicemapper is supported")
+	}
+
+	if len(info.DriverStatus) == 0 {
+		return nil, fmt.Errorf("DriverStatus not found")
+	}
+
+	stats := &DeviceMapperStats{}
+	for _, status := range info.DriverStatus {
+		value, err := humanize.ParseBytes(status[1])
+		if err != nil {
+			continue
+		}
+		switch status[0] {
+		case "Data Space Used":
+			stats.DataUsed = value
+		case "Data Space Available":
+			stats.DataAvailable = value
+		case "Data Space Total":
+			stats.DataTotal = value
+		case "Metadata Space Used":
+			stats.MetadataUsed = value
+		case "Metadata Space Available":
+			stats.MetadataAvailable = value
+		case "Metadata Space Total":
+			stats.MetadataTotal = value
+		}
```

Comment:
I'm not sure if other files in this package do it, but perhaps it could be a good idea to put a log.Verbose statement in the `default` branch so we log something when encountering unexpected metrics here.

Reply:
This is example of the key:values data here (from docker info)
```
 Storage Driver: devicemapper
  Pool Name: docker-259:1-1049197-pool
  Pool Blocksize: 65.54kB
  Base Device Size: 10.74GB
  Backing Filesystem: xfs
  Udev Sync Supported: true
  Data file: /dev/loop0
  Metadata file: /dev/loop1
  Data loop file: /var/lib/docker/devicemapper/devicemapper/data
  Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
  Data Space Used: 879MB
  Data Space Total: 107.4GB
  Data Space Available: 28GB
  Metadata Space Used: 18.64MB
  Metadata Space Total: 2.147GB
  Metadata Space Available: 2.129GB
  Thin Pool Minimum Free Space: 10.74GB
  Deferred Removal Enabled: true
  Deferred Deletion Enabled: true
  Deferred Deleted Device Count: 0
  Library Version: 1.02.146-RHEL7 (2018-01-22)
  ```
I thought it was going to be too verbose if we print from default since there are some metrics that will follow on this category always.
I would say that is expected to encounter unexpected metrics that is why i didn't log them